service/qos/service_level_controller.cc

```diff
@@ -82,19 +87,28 @@ future<> service_level_controller::drain() {
     if (this_shard_id() != global_controller) {
         return make_ready_future();
     }
-    // abort the loop of the distributed data checking if it is running
```

Comment:
Please convert to a coroutine before. If you like, you can post the coroutinization patches separately to get them merged earlier.

Reply:
I think it's okay to coroutinize it in this patch